sopel/tools/calculation.py

```diff
@@ -78,7 +81,7 @@ def _eval_node(self, node, timeout):
             return self.unary_ops[type(node.op)](operand)
 
         raise ExpressionEvaluator.Error(
-            "Ast.Node '%s' not implemented." % (type(node).__name__,))
+            "ast.Node '%s' not implemented." % (type(node).__name__,))
```

Comment:
nit: there is no `ast.Node`, it would be marginally better here (imo) to drop the expression-y spelling

```suggestion
            "AST node type '%s' not implemented." % (type(node).__name__,))
```

Reply:
Will leave this open, and apply after we've dropped Python 3.7. We already know that GH will need a kick in the pants to create a fresh merge after upstream changes, and this will be the perfect kick to give it. 😁

Reply:
Everything being an `ast.Constant` now anyway, it probably needs to use something other than `type(node)`. Maybe:

```suggestion
            "'%s' values are not supported." % (type(node.v).__name__,))
```

Reply:
Mmm, I agree with the generalization of the message, but I don't think we can rely on that , because not every node that might fail here is a `Constant`, and not all node types will have an attribute of this sort that we can use.

Maybe we want to split this into two cases? One that looks like what you drafted (but with `.value` rather than `.v`) for `Constant` nodes that gives a more specific report, and one for everything else that looks more like the existing report.

I.e. something like:

```python
        if isinstance(node, ast.Constant):
            msg = "'%s' values are not supported." % (type(node.value).__name__,)
        else:
            msg = "Node of type '%s' is not supported" % (type(node).__name__,)
        raise ExpressionEvaluator.Error(
            "ast.Node '%s' not implemented." % (type(node).__name__,))
```

Because I am in _"annoying nit/edge-case"_ mode, I think even this is suboptimal, because there are some narrower types of `BinOp, UnaryOp` that might not be supported, too? That one is probably easier to ignore, but maybe we want to push the narrowing `isinstance()` checks into the body of the relevant clauses, raise specific errors when those narrowing checks fail, and reword the catch-all case to be a suitably generic _"user's expression is on fire in a way we didn't anticipate"_ report. I'll make a separate suggestion for this quibble.

Reply:
Adapted this into an add-on commit w/co-author.